R/ParameterIdentification.R

```diff
@@ -731,6 +731,224 @@ ParameterIdentification <- R6::R6Class(
+      }
+      defaultGrid <- tibble::as_tibble(gridList)
+
+      profileList <- vector(mode = "list", length = np)
+      for (idx in seq_along(private$.piParameters)) {
+        # the names of the parameters are extracted from the first available path
+        parameterName <- private$.piParameters[[idx]]$parameters[[1]]$path
+        grid <- seq(from = lower[[idx]], to = upper[[idx]], length.out = gridSize)
+        currentGrid <- defaultGrid
+        currentGrid[[parameterName]] <- grid
+        # creates a tibble with the column name from the `parameterName` variable
+        profileList[[idx]] <- tibble::tibble(!!parameterName := grid)
+        profileList[[idx]][["ofv"]] <- purrr::pmap_dbl(currentGrid, function(...) {
+          private$.targetFunction(c(...))$model
+        })
+        names(profileList)[[idx]] <- parameterName
+      }
+
+      # Mark that the batches have been initialized and restore simulation state
+      # Note: we can't use `simulationState` variable in the condition
+      # because it might be not defined
+      if (private$.needBatchInitialization){
+        .restoreSimulationState(private$.simulations, simulationState)
+        private$.needBatchInitialization <- FALSE
+      }
+
+      return(profileList)
+    },
+
+    #' @description
+    #' Plot the profiles of the objective function calculated by the calculateOFVProfiles() method.
+    #' @param profiles A list of tibbles, with each tibble containing two columns. The first column should contain
+    #' several values for the parameter, and the second column, called `ofv`, should contain the matching objective function value.
+    #' @return A list of ggplot2 objects, one for each parameter.
+    plotOFVProfiles = function(profiles) {
+      if (missing(profiles)) {
+        stop("Supply the result of the calculateOFVProfiles() method as the argument to the plotOFVProfiles() method.")
+      }
+      plotList <- vector(mode = "list", length = length(profiles))
+      for (idx in seq_along(profiles)) {
+        parameterName <- names(profiles)[[idx]]
+        plotList[[idx]] <- ggplot2::ggplot(data = profiles[[idx]],
+                                           ggplot2::aes(x = .data[[parameterName]], y = .data[["ofv"]])) +
+          ggplot2::theme_bw() +
+          ggplot2::geom_point(ggplot2::aes(col = 1/ofv)) +
+          ggplot2::labs(x = parameterName, y = "OFV") +
+          ggplot2::scale_color_viridis_c() +
+          ggplot2::guides(col = "none")
+      }
+      return(plotList)
+    },
+
+    #' @description
+    #' Plot a heatmap of the objective function calculated by the gridSearch() method
+    #' @param grid A tibble with three columns. The first two columns should contain parameter values
+    #' for a 2-parameter optimization problem, and the `ofv` column should contain the matching objective function values.
+    #' @return A ggplot2 object.
+    plotGrid = function(grid) {
```

Comment:
What happens for number of parameters != 2?

Maybe rename `plot2DOFVGrid`? Also I think `plotOFVProfiles` and this function should be separate functions in a `utilities`, because they do not use the Parameter Identification object. In theory, one could you them with inputs (grid, profiles) generated somehow differntly.